network/adaptive_api.go

```diff
@@ -0,0 +1,45 @@
+package network
```

Comment:
how about creating a package for capabilities, then we could get rid of all these long names?

Reply:
Ok so merge this then I move it to separate package please?

Reply:
is this such a big overhead to do now @nolash? we're talking 102 lines delta....